lang/java/tools/src/test/java/org/apache/avro/tool/TestConcatTool.java

```diff
@@ -136,6 +137,38 @@ public void testConcat() throws Exception {
   }
 
   @Test
+  @Ignore
+  // theoretically this test should test the append scenario
+  // the problem is, FileSystem.append does not work for local filesystems
+  // (note: or maybe doing dev on Windows is the problem??)
+  // the best way to test is to write to HDFS and compare append output to the
+  // fresh output file generated by the plain concat operation writing to HDFS
```

Comment:
I'm going to start by apologizing. I did not realize there was such a dearth of information on using MiniDFSCluster. :/

Adding this will require a test scope dependency on the hadoop-hdfs test-jar, just fyi.

The best I have to offer on actually using MiniDFSCluster are pointers to others using it.

* HBase using MiniDFSCluster in their testing utility: [HBaseTestingUtility](https://github.com/apache/hbase/blob/rel/1.2.1/hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java#L565) (note that this only uses the direct constructor, which is hard to follow, but it has the essential pattern: create minidfscluster, start it, retrieve a FileSystem object).
* The CLI tool for making mini cluster instances that ships with Hadoop has a good use of the proper Builder pattern of creating a minidfs cluster [MiniDFSClusterManager](https://github.com/apache/hadoop/blob/release-2.5.1/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/test/MiniDFSClusterManager.java#L139)
* The actual Builder class is the best way to see the available options, but sadly Hadoop does not publish javadocs for test classes, so we have to look at source. [MiniDFSCluster](https://github.com/apache/hadoop/blob/release-2.5.1/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/MiniDFSCluster.java#L140).